### PR TITLE
properly handle scalar index variable. Fix ROOT-10557.

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: tree/treeplayer/src/TTreeFormula.cxx [hunk 1]
```diff
@@ -363,14 +363,18 @@ void TTreeFormula::DefineDimensions(Int_t code, Int_t size,
...
       switch (index_multiplicity) {
-         case -1:
          case  0:
+            scalarindex = true;
+            vsize = 1;
+            break;
+         case -1:
          case  2:
             vsize = indexvar->GetNdata();
             break;
```
Comment (hageboeck) [inline tree/treeplayer/src/TTreeFormula.cxx]: This looks like a case for an `if` and/or a ternary operator. Personally, I find the fall throughs harder to read.

File: tree/treeplayer/src/TTreeFormula.cxx [hunk 2]
```diff
@@ -382,7 +386,7 @@ void TTreeFormula::DefineDimensions(Int_t code, Int_t size,
 
    fCumulSizes[code][fNdimensions[code]] = size;
 
-   if ( fIndexes[code][fNdimensions[code]] < 0 ) {
+   if ( !scalarindex && fIndexes[code][fNdimensions[code]] < 0 ) {
```
Comment (hageboeck) [inline tree/treeplayer/src/TTreeFormula.cxx]: It's hard to say from reading the code, but it looks like `scalarindex` is equivalent to `vsize == 1`. Maybe it's unnecessary.

Comment (pcanal) [reply tree/treeplayer/src/TTreeFormula.cxx]: vsize==1 could also be a std::vector that happen to have one element in this entry.

**Pull request merged**